lxd/patches.go

```diff
@@ -3714,3 +3713,65 @@ func patchUpdateFromV30(_ *sql.Tx) error {
 
 	return nil
 }
+
+//  renames any config incorrectly set config file entries due to the lvm.thinpool_name typo
+func patchThinpoolTypoFix(name string, d *Daemon) error {
+	tx, err := d.cluster.Begin()
```

Comment:
Instead of manually creating a transaction that way, the entire part of the function interacting with the database should be in a `err = d.cluster.Transaction(func(tx *db.ClusterTx) error {` type block.

That way things will properly revert on failure rather than potentially keeping the database in a locked state.